src/Database/Redis/Core.hs

```diff
@@ -2,7 +2,7 @@
     MultiParamTypeClasses, FunctionalDependencies, FlexibleInstances, CPP #-}
 
 module Database.Redis.Core (
-    Connection, connect,
+    Connection(..), connect,
```

Comment:
I propose keeping this type abstract, while providing `unConnection` deconstructor which would be explicitly said to be internal.